plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/TimestampCoercer.java

```diff
@@ -155,4 +163,16 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
             }
         }
     }
+
+    private static LocalDateTime toLocalDateTime(LongTimestamp timestamp)
+    {
+        long epochSecond = floorDiv(timestamp.getEpochMicros(), MICROSECONDS_PER_SECOND);
+        long microsFraction = floorMod(timestamp.getEpochMicros(), MICROSECONDS_PER_SECOND);
+        // Hive timestamp has nanoseconds precision, so no truncation here
+        long nanosFraction = (microsFraction * NANOSECONDS_PER_MICROSECOND) + (timestamp.getPicosOfMicro() / PICOSECONDS_PER_NANOSECOND);
+        if (epochSecond < START_OF_MODERN_ERA_SECONDS) {
+            throw new TrinoException(HIVE_INVALID_TIMESTAMP_COERCION, "Coercion on historical dates is not supported");
+        }
+        return LocalDateTime.ofEpochSecond(epochSecond, toIntExact(nanosFraction), UTC);
+    }
```

Comment:
Why not use the logic from `io.trino.operator.scalar.timestamp.TimestampToDateCast` which, fully-expanded, is just:

```
return java.lang.Math.floorDiv(timestamp.getEpochMicros(), io.trino.type.DateTimes.MICROSECONDS_PER_DAY)
```